packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanDetail/AccordianLogs.tsx

```diff
@@ -52,34 +52,45 @@ export default function AccordianLogs({
   let arrow: React.ReactNode | null = null;
   let HeaderComponent: 'span' | 'a' = 'span';
   let headerProps: object | null = null;
-  const [showOutOfRangeLogs, setShowOutOfRangeLogs] = React.useState(false);
+  const [hasAutoOpened, setHasAutoOpened] = React.useState(false);
+  const [expandedOnce, setExpandedOnce] = React.useState(false);
+
+  React.useEffect(() => {
+    if (!hasAutoOpened && !isOpen && logs && logs.length > 0 && onToggle) {
+      setHasAutoOpened(true);
+      onToggle();
+    }
+  }, [hasAutoOpened, isOpen, logs, onToggle]);
 
   const inRangeLogs = React.useMemo(() => {
     const viewStartAbsolute = timestamp + currentViewRangeTime[0] * traceDuration;
     const viewEndAbsolute = timestamp + currentViewRangeTime[1] * traceDuration;
     return logs.filter(log => log.timestamp >= viewStartAbsolute && log.timestamp <= viewEndAbsolute);
   }, [logs, timestamp, currentViewRangeTime, traceDuration]);
 
-  const logsToDisplay = showOutOfRangeLogs ? logs : inRangeLogs;
-  const displayedCount = logsToDisplay.length;
-  const inRangeCount = inRangeLogs.length;
+  const autoExpandCount = 3;
+  const logsToDisplay = expandedOnce ? inRangeLogs : inRangeLogs.slice(0, autoExpandCount);
+
   const totalCount = logs.length;
+  const hasMoreLogs = inRangeLogs.length > autoExpandCount && !expandedOnce;
 
-  let title = `Logs (${displayedCount})`;
-  let toggleLink: React.ReactNode = null;
+  const title = `Logs (${totalCount})`;
+  let showMoreLink: React.ReactNode = null;
 
-  if (!showOutOfRangeLogs && inRangeCount < totalCount) {
-    title = `Logs (${inRangeCount} of ${totalCount})`;
-    toggleLink = (
-      <button type="button" className="AccordianLogs--toggle" onClick={() => setShowOutOfRangeLogs(true)}>
-        show all
-      </button>
-    );
-  } else if (showOutOfRangeLogs && inRangeCount < totalCount) {
-    title = `Logs (${totalCount})`;
-    toggleLink = (
-      <button type="button" className="AccordianLogs--toggle" onClick={() => setShowOutOfRangeLogs(false)}>
-        show in range
+  if (hasMoreLogs) {
```

Comment:
hasMoreLogs defined far away and never reused
```suggestion
  if (inRangeLogs.length > autoExpandCount && !expandedOnce) {
```